source/components/ThemeProvider.js

```diff
@@ -1,74 +1,54 @@
-import React, { Component } from 'react';
-import { object, string, element } from 'prop-types';
+import React, { Component } from "react";
+import { ThemeContext } from "../themes/ThemeContext";
+import { object, string, element } from "prop-types";
 
 // external libraries
-import _ from 'lodash';
+import _ from "lodash";
 
 // imports the Root Theme API object which specifies the shape
 // of a complete theme for every component in this library, used in this.composeLibraryTheme
-import ROOT_THEME_API from '../themes/API';
+import ROOT_THEME_API from "../themes/API";
 
 // internal utility functions
-import { composeTheme } from '../utils';
+import { composeTheme } from "../utils";
 
 class ThemeProvider extends Component {
   static propTypes = {
-    children: element,
     theme: object,
     themeOverrides: object // custom css/scss from user that adheres to shape of ROOT_THEME_API
   };
 
   static defaultProps = {
-    theme: { ...ROOT_THEME_API },
     themeOverrides: {}
   };
 
-  static childContextTypes = {
-    theme: object
-  };
-
   constructor(props) {
     super(props);
 
     const { theme, themeOverrides } = props;
 
     this.state = {
-      composedTheme: this.composeLibraryTheme(
-        theme,
-        themeOverrides,
-        ROOT_THEME_API
-      )
-    };
-  }
-
-  getChildContext() {
-    const { composedTheme } = this.state;
-
-    return {
-      theme: { ...composedTheme }
+      theme: this.composeLibraryTheme(theme, themeOverrides, ROOT_THEME_API),
+      ROOT_THEME_API
     };
   }
 
-  // prevents frequent rerenders of ThemeProvider's children by
-  // checking if theme and/or themeOverrides props have changed
-  componentWillReceiveProps(nextProps) {
-    const { theme, themeOverrides } = nextProps;
-
-    const changedProps = _.pickBy(
-      { theme, themeOverrides },
-      (value, key) => this.props[key] !== value
-    );
-
-    if (Object.keys(changedProps).length > 0) {
-      this.setState({
-        composedTheme: this.composeLibraryTheme(
-          theme,
-          themeOverrides,
-          ROOT_THEME_API
-        )
-      });
-    }
-  }
+  // // prevents frequent rerenders of ThemeProvider's children by
+  // // checking if theme and/or themeOverrides props have changed
+  // componentWillReceiveProps(nextProps) {
+  //   const { theme, themeOverrides } = nextProps;
+  //
+  //   const changedProps = _.pickBy(
+  //     { theme, themeOverrides },
+  //     (value, key) => this.props[key] !== value
+  //   );
+  //
+  //   if (Object.keys(changedProps).length > 0) {
+  //     this.setState({
+  //       composedTheme: this.composeLibraryTheme(theme, themeOverrides, ROOT_THEME_API)
+  //     });
+  //   }
+  // }
```

Comment:
Is this still needed? If not it's probably better to just remove it than to leave it commented out … or add a `// TODO: …` comment above to explain your thought process about this part is.